test/common/index.js

```diff
@@ -802,3 +802,32 @@ exports.hijackStdout = hijackStdWritable.bind(null, 'stdout');
 exports.hijackStderr = hijackStdWritable.bind(null, 'stderr');
 exports.restoreStdout = restoreWritable.bind(null, 'stdout');
 exports.restoreStderr = restoreWritable.bind(null, 'stderr');
+
+// Check if the executable links to the shared lib node or not.
+exports.isSharedLibNode = process.config.variables.node_shared === true;
+
+// If node executable is linked to shared lib, need to take care about the
+// shared lib path.
+exports.addLibraryPath = function() {
+  if (!process.config.variables.node_shared) {
+    return;
+  }
+  process.env.LD_LIBRARY_PATH =
+    (process.env.LD_LIBRARY_PATH ?
+      process.env.LD_LIBRARY_PATH + path.delimiter : '') +
+    path.join(path.dirname(process.execPath), 'lib.target');
+  // For AIX.
+  process.env.LIBPATH =
+    (process.env.LIBPATH ?
+      process.env.LIBPATH + path.delimiter : '') +
+    path.join(path.dirname(process.execPath), 'lib.target');
+  // For Mac OSX.
+  process.env.DYLD_LIBRARY_PATH =
+    (process.env.DYLD_LIBRARY_PATH ?
+      process.env.DYLD_LIBRARY_PATH + path.delimiter : '') +
+    path.dirname(process.execPath);
+  // For Windows.
+  process.env.PATH =
+    (process.env.PATH ? process.env.PATH + path.delimiter : '') +
+    path.dirname(process.execPath);
+};
```

Comment:
Since this is only going to be called rarely, I feel like it would be better to move this to a separate file.

Reply:
move to a file which both `test-child-process-fork-exec-path` and `test-module-loading-globalpaths` can access? not familiar with the directory structure of the testing suites, any suggestion?

Reply:
There is for example `test/common/countdown.js`. Similar to that you can just add another file.

Reply:
@BridgeAR I moved the function to a new file: `shared-lib-util.js`